src/core/namespaceValidator.js

```diff
@@ -0,0 +1,36 @@
+import Exports from '../ExportMap';
+
+export default function namespaceValidator(node, namespaces, onComputed, { onNamespaceNotFound = undefined, onDeprecation = undefined } = {}) {
+  // go deep
+  let namespace = namespaces.get(node.object.name);
+  const namepath = [node.object.name];
+  // while property is namespace and parent is member expression, keep validating
+  while (namespace instanceof Exports && node.type === 'MemberExpression') {
+
+    if (node.computed) {
+      if (onComputed(node) === 'return') {
+        return;
+      }
+    }
+
+    if (onNamespaceNotFound) {
+      if (onNamespaceNotFound(node, namespace, namepath) === 'return') {
+        return;
+      }

```

Comment:
I re-produced exact flow as in `namespace` rules: https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/namespace.js. Lines 124-140.

Line 139 of original namespace rule has `break` instead of return, but because nothing comes after `while` loop, `break` does the same thing as `return`?

Reply:
Are there any cases of calling `onNamespaceNotFound` where we *don't* want to break/return?

Reply:
Yes. `onNamespaceNotFound` is only used in namespace rule.

If we can find node's name in namespace map, then but just proceed execution. We only return/break when we cannot find node's name in namespace map.

Below is onNamespaceNotFound` in `namespace` rule, we pass what's below to our helper function.
```
        function onNamespaceNotFound(context) {
          return function inner(node, namespace, namepath) {
            if (!namespace.has(node.property.name)) {
              context.report(
                node.property,
                makeMessage(node.property, namepath),
              );
              return 'return';
            }
          };
        }
```

Reply:
gotcha, hmm.

not a fan of returning the string "return", maybe it could be a true or something, but returning a sentinel value seems to be the only solution.

ahhhh wait! what if the `inner` callback took a fourth argument, that meant "return", for it to return? that way "what kind of value if is" is up to the helper implementation

Reply:
That's brilliant! I implemented it here: https://github.com/azyzz228/eslint-plugin-import/commit/5985656c1dcd2f738f3fb222cce772a9aa58d77f